Userland/Libraries/LibGUI/Dialog.cpp

```diff
@@ -28,18 +28,25 @@ int Dialog::exec()
     VERIFY(!m_event_loop);
     m_event_loop = make<Core::EventLoop>();
 
-    auto desktop_rect = Desktop::the().rect();
+    bool has_parent_window = parent() && is<Window>(parent());
+
+    size_t screen_index = Desktop::the().main_screen_index();
+    if (has_parent_window) {
+        screen_index = static_cast<Window*>(parent())->containing_screen_index();
+    }
+    auto screen_rect = Desktop::the().rects()[screen_index];
+
     auto window_rect = rect();
 
-    auto top_align = [](Gfx::Rect<int>& rect) { rect.set_y(32); };
-    auto bottom_align = [this, desktop_rect](Gfx::Rect<int>& rect) { rect.set_y(desktop_rect.height() - Desktop::the().taskbar_height() - height() - 12); };
+    auto top_align = [screen_rect](Gfx::Rect<int>& rect) { rect.set_y(screen_rect.y() + 32); };
+    auto bottom_align = [this, screen_rect](Gfx::Rect<int>& rect) { rect.set_y(screen_rect.y() + screen_rect.height() - Desktop::the().taskbar_height() - height() - 12); };
```

Comment:
This was already wrong, the taskbar is only displayed if the screen you're centering on is the same as `Desktop::the().main_screen_index()`